segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/ContainerMetadataUpdateTransaction.java

```diff
@@ -114,13 +114,8 @@
         this.newTruncationPoints = new ArrayList<>();
         this.newSegments = new HashMap<>();
         this.newSegmentNames = new HashMap<>();
-        if (baseMetadata.isRecoveryMode()) {
-            this.newSequenceNumber = ContainerMetadata.INITIAL_OPERATION_SEQUENCE_NUMBER;
-        } else {
-            this.newSequenceNumber = Long.MIN_VALUE;
-        }
-
         this.sealed = false;
+        resetNewSequenceNumber();
```

Comment:
This `newSequenceNumber` is only set (and should only be used) during recovery, when we need to restore the ContainerMetadata's SequenceNumber to what it was before. In normal operations this does not get touched. If you check out the methods that set this value, they first check that the metadata is not in recovery mode. 

The bug was that I was using this value when I was assigning new segment ids. Since in that case this value was always Long.MIN_VALUE, there was a good chance of segment id collision (if a previous segment got evicted or deleted), hence the bug described in the original issue (not the PR).